waterspout_api/views.py

```diff
@@ -35,6 +40,96 @@ def get_user_info_dict(user, token):
 		}
 
 
+class GetPasswordReset(APIView):
```

Comment:
Flagging for myself that I'd like to test this for security, but I think your test cases cover most of what's needed. I'll need to pull it to my local machine and give it a try.